mtp_prisoner_location_admin/apps/prisoner_location_admin/views.py

```diff
@@ -0,0 +1,24 @@
+from django.views.generic.edit import FormView
+from django.core.urlresolvers import reverse_lazy
+from django import forms
+
+from .forms import LocationFileUploadForm
+
+
+class LocationFileUploadView(FormView):
+    template_name = 'prisoner_location_admin/location_file_upload.html'
+    form_class = LocationFileUploadForm
+    success_url = reverse_lazy('dashboard')
+
+    def get_form_kwargs(self):
+        kwargs = super(LocationFileUploadView, self).get_form_kwargs()
+        kwargs['request'] = self.request
+        return kwargs
+
+    def form_valid(self, form):
+        try:
+            form.update_locations()
+            return super(LocationFileUploadView, self).form_valid(form)
+        except forms.ValidationError as e:
+            form.add_error(None, e)
```

Comment:
Are you referring to the fact that normally raising a ValidationError is caught and converted to a form error automatically? If so, it's because that only happens when it occurs in the 'clean' phase of form processing, whereas this is in the method that is called after the 'clean' phase is completed successfully.

The reason for this is that I didn't want to add validation logic (beyond just the file format) in the client app, so instead any problems with the actual data are found in the API processing and returned as a 400, then reraised as a ValidationError (which is manually caught and displayed as a form error).